### PR TITLE
unit test for analyze_energy

### LINKED ISSUE #1312: Integration test for energy calculation
Total energy and individual components of analysis.energy()

### PR DESCRIPTION
**Pull request by davidsean**
Fixes #1312

Description of changes:
 - 

PR Checklist
------------
 - [ ] Tests?
   - [ ] Interface
   - [ ] Core 
 - [ ] Docs?

### CODE REVIEW

File: testsuite/analyze_energy.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
+from __future__ import print_function
+import sys
+import unittest as ut
+import numpy as np
+import espressomd
+#from espressomd.interactions import *
```
Comment (KaiSzuttor) [inline testsuite/analyze_energy.py]: Please remove

File: testsuite/analyze_energy.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+from espressomd.interactions import HarmonicBond
+
+@ut.skipIf(not espressomd.has_features("LENNARD_JONES"),"Skipped because LENNARD_JONES turned off.")
+
```
Comment (KaiSzuttor) [inline testsuite/analyze_energy.py]: Please remove empty line

File: testsuite/analyze_energy.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    system.bonded_inter.add(harmonic)
+        
+    def test_kinetic(self):
+        self.system.part[0].pos = [1, 2, 2]
+        self.system.part[1].pos = [5, 2, 2]
+        self.system.part[0].v = [3, 4, 5]
+        self.system.part[1].v = [0, 0, 0]
+        # initial, single moving particle
+        energy = self.system.analysis.energy()
+        self.assertTrue( np.allclose(energy["total"], 25.))
```
Comment (KaiSzuttor) [inline testsuite/analyze_energy.py]: Please run autopep8 -i on that file to get rid of the unnecessary blanks

File: testsuite/analyze_energy.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+import sys
+import unittest as ut
+import numpy as np
+import espressomd
+#from espressomd.interactions import *
+from espressomd.interactions import HarmonicBond
+
+@ut.skipIf(not espressomd.has_features("LENNARD_JONES"),"Skipped because LENNARD_JONES turned off.")
+
+class test_analyze_energy(ut.TestCase):
```
Comment (KaiSzuttor) [inline testsuite/analyze_energy.py]: Recommended is camelcase namestyle for classes

File: testsuite/analyze_energy.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        self.assertTrue( np.allclose(energy["non_bonded"], 1. + 1.))
+        self.system.part[3].remove()
+        self.system.part[4].remove()
+        self.system.part[0].delete_all_bonds()
+
+
+
+if __name__ == "__main__":
+    print("Features: ", espressomd.features())
+
```
Comment (KaiSzuttor) [inline testsuite/analyze_energy.py]: remove empty line

File: testsuite/analyze_energy.py [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+
+if __name__ == "__main__":
+    print("Features: ", espressomd.features())
+    
+    suite = ut.TestLoader().loadTestsFromTestCase(test_analyze_energy)
+    ut.TextTestRunner(verbosity=2).run(suite)
+
+    #ut.main()
```
Comment (KaiSzuttor) [inline testsuite/analyze_energy.py]: remove that line

**Review by KaiSzuttor — CHANGES_REQUESTED** [review]

File: testsuite/analyze_energy.py [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+@ut.skipIf(not espressomd.has_features("LENNARD_JONES"),"Skipped because LENNARD_JONES turned off.")
+
+class test_analyze_energy(ut.TestCase):
+    system = espressomd.System()
+    box_l = 10.0
+
+    system.box_l = [box_l, box_l, box_l]
+    system.cell_system.skin = 0.4
+    system.time_step = 0.01
+    system.non_bonded_inter[0, 0].lennard_jones.set_params(
```
Comment (KaiSzuttor) [inline testsuite/analyze_energy.py]: Move that part into the setUp method of unittest

**Review by fweik — APPROVED** [review]

**Comment by KaiSzuttor** [comment]
test is failing...

File: testsuite/analyze_energy.py [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        self.assertTrue(np.allclose(energy["total"], 50.))
+        self.assertTrue(np.allclose(energy["kinetic"], 50.))
+        self.assertTrue(np.allclose(energy["bonded"], 0.))
+        self.assertTrue(np.allclose(energy["non_bonded"], 0.))
+
+    def test_non_bonded(self):
+        self.system.part[0].pos = [1, 2, 2]
+        self.system.part[1].pos = [2, 2, 2]
+        energy = self.system.analysis.energy()
+        self.assertTrue(np.allclose(energy["total"], 1.))
```
Comment (KaiSzuttor) [inline testsuite/analyze_energy.py]: That fails because you  still have particles from tests before.

File: testsuite/analyze_energy.py [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+        self.assertTrue(np.allclose(energy["kinetic"], 0.))
```
Comment (KaiSzuttor) [inline testsuite/analyze_energy.py]: please use self.assertAlmostEqual(a, b, delta=1e-7) or similar

**Review by KaiSzuttor — APPROVED** [review]

**Pull request merged**